src/main/java/seedu/address/logic/parser/ClientNoteEditCommandParser.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ClientNoteAddCommand;
+import seedu.address.logic.commands.ClientNoteEditCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.note.Note;
+import seedu.address.model.note.TagNoteMap;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new ClientNoteEditCommandParser object.
+ */
+public class ClientNoteEditCommandParser implements Parser<ClientNoteEditCommand> {
+
+    private final TagNoteMap tagNoteMap;
+
+    /**
+     * Initializes a {@code ClientNoteEditCommandParser} with the {@code tagNoteMap} object.
+     */
+    public ClientNoteEditCommandParser(TagNoteMap tagNoteMap) {
+        requireNonNull(tagNoteMap);
+        this.tagNoteMap = tagNoteMap;
+    }
+
+    @Override
+    public ClientNoteEditCommand parse(String restOfCommand) throws ParseException {
+        requireNonNull(restOfCommand);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(restOfCommand, PREFIX_NOTE, PREFIX_TAG);
+        // NEED TO PASS IN NOTE STRING AT LEAST, TAGS ARE OPTIONAL
+        if (!arePrefixesPresent(argMultimap, PREFIX_NOTE) || argMultimap.getPreamble().isEmpty()) {
```

Comment:
No need to use this arePrefixesPresent function if we only have one prefix to check. Also should be either note string or tag prefix present

Reply:
added to issue #263